src/main/java/org/elasticsearch/search/aggregations/metrics/stats/extended/InternalExtendedStats.java

```diff
@@ -104,6 +124,33 @@ public double getStdDeviation() {
     }
 
     @Override
+    public double getStdDeviationBound(Bounds bound) {
+        if (bound.equals(Bounds.UPPER)) {
+            return getAvg() + (getStdDeviation() * sigma);
+        } else {
+            return getAvg() - (getStdDeviation() * sigma);
+        }
+    }
+
+    @Override
+    public double getStdError() {
+        return getStdDeviation() / Math.sqrt(count);
+    }
+
+    @Override
+    public double getConfidenceBound(Bounds bound) {
+        if (confidenceInterval == -1) {
+            return Double.NaN;
```

Comment:
Second thought, you're right...an exception will be more friendly.  It at least tells the user why the value isn't provided, rather than a nebulous NaN.  Will make the change.